com.unity.cinemachine/Editor/Timeline/CinemachineShotEditor.cs

```diff
@@ -4,59 +4,138 @@
-        /// <param name="excluded">Add the names to this list</param>
-        protected override void GetExcludedPropertiesInInspector(List<string> excluded)
+        void OnEnable() => Undo.undoRedoPerformed += UpdateComponentEditors;
+        void OnDisable() => Undo.undoRedoPerformed -= UpdateComponentEditors;
+
+        public override VisualElement CreateInspectorGUI()
         {
-            base.GetExcludedPropertiesInInspector(excluded);
-            excluded.Add(FieldPath(x => x.VirtualCamera));
+            var m_ParentElement = new VisualElement();
+
+            // Auto-create shots
+            var toggle = m_ParentElement.AddChild(new Toggle(CinemachineTimelinePrefs.s_AutoCreateLabel.text) 
+            { 
+                tooltip = CinemachineTimelinePrefs.s_AutoCreateLabel.tooltip,
+                value = CinemachineTimelinePrefs.AutoCreateShotFromSceneView.Value
+            });
+            toggle.AddToClassList(InspectorUtility.kAlignFieldClass);
+            toggle.RegisterValueChangedCallback((evt) => CinemachineTimelinePrefs.AutoCreateShotFromSceneView.Value = evt.newValue);
+
+            // Cached scrubbing
+            var row = m_ParentElement.AddChild(new VisualElement { style = { flexDirection = FlexDirection.Row }});
+            var cacheToggle = row.AddChild(new Toggle(CinemachineTimelinePrefs.s_ScrubbingCacheLabel.text) 
+            { 
+                tooltip = CinemachineTimelinePrefs.s_ScrubbingCacheLabel.tooltip,
+                value = CinemachineTimelinePrefs.UseScrubbingCache.Value,
+                style = { flexGrow = 0 }
+            });
+            cacheToggle.AddToClassList(InspectorUtility.kAlignFieldClass);
+
+            row.Add(new Label { text = "(experimental)", style = { flexGrow = 1, alignSelf = Align.Center } });
+            var clearCacheButton = row.AddChild(new Button 
+            {
+                text = "Clear",
+                style = { flexGrow = 0, alignSelf = Align.Center }
+            });
+            clearCacheButton.RegisterCallback<ClickEvent>((evt) => TargetPositionCache.ClearCache());
+            clearCacheButton.SetEnabled(CinemachineTimelinePrefs.UseScrubbingCache.Value);
+            cacheToggle.RegisterValueChangedCallback((evt) => 
+            {
+                CinemachineTimelinePrefs.UseScrubbingCache.Value = evt.newValue;
+                clearCacheButton.SetEnabled(evt.newValue);
+            });
+
+            m_ParentElement.Add(new PropertyField(serializedObject.FindProperty(() => Target.VirtualCamera)));
+            m_ParentElement.Add(new PropertyField(serializedObject.FindProperty(() => Target.DisplayName)));
+
+            UpdateComponentEditors();
+            return m_ParentElement;
         }
 
-        void OnDisable()
+        void UpdateComponentEditors()
         {
-            DestroyComponentEditors();
+            if (m_ParentElement == null || serializedObject == null)
```

Comment:
Because this is a callback called from UndoRedoPerformed, I'm being extra careful.  I've had issues in the past where these kinds of callbacks were invoked after an object was deleted